### PR TITLE
Match creation, setup, joining and spectate

### PR DESCRIPTION
**Pull request by JaviTrek**
# Match creation, setup, joining and spectate are all functional now!
image

## Match Creation
Now we can create matches and have a default CO + army. Matches a player is part of show on the your matches tab, while matches made by other players that are still waiting for a player appear under the "join" tab and we also have a tab to spectate matches that already have 2 players which aren't the current user.

## Match Joining
As simple as it sounds, now you can join a match create by another player.

## Match Setup
Changing COs, army, readying up/unreadying, now all available in our match area! And all of these data is sent straight into the db so its not just changes in the frontend.

## Enter Match
Now when you enter a match, it checks that in fact, the match has started and currently only shows the "enter match" button if the match is set to live!

### CODE REVIEW

File: public/img/nations/black-hole.gif
[the remark this answers is not in the document]
Comment (JaviTrek) [reply public/img/nations/black-hole.gif:1]: @CliveGlitch "Also, the images are of lower quality in comparison with what we had." You mean as in the pixel art vs the smooth ones or you are saying the images/duplicates I deleted were of higher quality?

**Comment by JaviTrek** [comment]
@CliveGlitch waiting on making sure you are done with any commits or PRs before merging into main, let me know when you are good! Thanks again for the help!

File: public/img/nations/black-hole.gif
[the remark this answers is not in the document]
Comment (CliveGlitch) [reply public/img/nations/black-hole.gif:1]: image
image
Yeah, although I don't think there is a HQ image for Black Hole

[the remark this answers is not in the document]
Comment (CliveGlitch) [reply public/img/nations/black-hole.gif:1]: image
image

Weirdly enough the images are gifs, and we'll probably gonna change them later.

[the remark this answers is not in the document]
Comment (CliveGlitch) [reply public/img/nations/black-hole.gif:1]: Never mind, I just saw what you meant with pixel art and how you used it haha sorry.
image

**Review by CliveGlitch — APPROVED** [review]
Glad to help, I'm done with the review and any changes. 
If you have any question feel free, and I hope the best for the project.

**Pull request merged**